include/sparrow/fixed_size_layout.hpp

```diff
@@ -0,0 +1,369 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or mplied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <algorithm>
+#include <concepts>
+#include <cstdint>
+#include <cassert>
+
+#include "sparrow/array_data.hpp"
+#include "sparrow/iterator.hpp"
+#include "sparrow/buffer.hpp"
+#include "sparrow/dynamic_bitset.hpp"
+
+namespace sparrow
+{
+    /**
+     * An iterator for `fixed_size_layout` operating on contiguous data only.
+     *
+     * @tparam T The type of the elements in the layout's data buffer.
+     * @tparam is_const A boolean indicating whether the iterator is const.
+     *
+     * @note This class is copyable and movable.
+     */
+    template <class T, bool is_const>
+    class fixed_size_layout_value_iterator
+        : public iterator_base
+        <
+            fixed_size_layout_value_iterator<T, is_const>,
+            mpl::constify_t<T, is_const>,
+            std::contiguous_iterator_tag
+        >
+    {
+    public:
+
+        using self_type = fixed_size_layout_value_iterator<T, is_const>;
+        using base_type = iterator_base
+        <
+            self_type,
+            mpl::constify_t<T, is_const>,
+            std::contiguous_iterator_tag
+        >;
+        using pointer = typename base_type::pointer;
+
+        fixed_size_layout_value_iterator() = default;
+        explicit fixed_size_layout_value_iterator(pointer p);
+
+    private:
+        using reference = typename base_type::reference;
+        using difference_type = typename base_type::difference_type;
+        using size_type = std::size_t;
+
+        reference dereference() const;
+        void increment();
+        void decrement();
+        void advance(difference_type n);
+        difference_type distance_to(const self_type& hs) const;
+        bool equal(const self_type& rhs) const;
+        bool less_than(const self_type& rhs) const;
+
+        pointer m_pointer = nullptr;
+
+        friend class iterator_access;
+    };
+
+    /**
+     * A contiguous layout for fixed size types.
+     *
+     * This class provides a contiguous layout for fixed size types, such as `uint8_t`, `int32_t`, etc.
+     * It iterates over the first buffer in the array_data, and uses the bitmap to skip over null.
+     * The bitmap is assumed to be present in the array_data.
+     *
+     * @tparam T The type of the elements in the layout's data buffer.
+     *           A fixed size type, such as a primitive type.
+     *
+     * @note This class is not thread-safe, exception-safe, copyable, movable, equality comparable.
```

Comment:
Again that's the default in C++ so I dont get why this is noted here?

About "exception safe" I'm not sure what you mean here, [exception safety is a scale](https://en.wikipedia.org/wiki/Exception_safety#Classification) so if it's not exception-safe "no guarantees are made" but we didnt specially designed with that in mind so to me it shouldnt even be mentionned.